src/syntax/test-rules.rkt

```diff
@@ -29,13 +29,18 @@
 (define (check-rule-correct test-rule ground-truth)
   (match-define (rule name p1 p2 itypes otype) test-rule)
   (define fv (dict-keys itypes))
-  (*var-reprs* (for/list ([(v t) (in-dict itypes)]) (cons v (get-representation* t))))
-  (define repr (get-representation* otype))
+  (*var-reprs* (for/list ([(v t) (in-dict itypes)]) (cons v (get-representation t))))
+  (define var-precs (for/list ([(v t) (in-dict itypes)]) (cons v t)))
+  (define repr (get-representation otype))       
+
+  ;; Some rule names are generated by adding a _<num> at the end of the base name. Extract the base name
+  ;; so *conditions* doesn't get confused
+  (define base-name (string->symbol (first (string-split (symbol->string name) "_"))))
```

Comment:
Nah man, just duplicate the conditions. And desugar them too. It's test code, we don't have to pretty it up for anyone.